cli/src/key_pair_storage.rs

```diff
@@ -150,17 +154,26 @@ fn get_or_create_path() -> Result<PathBuf, Error> {
     Ok(path_buf)
 }
 
+/// Build the path to the given file name under [dir_ready()].
 fn build_path(filename: &str) -> Result<PathBuf, Error> {
-    let dir = dir()?;
+    let dir = dir_ready()?;
     let path = dir.join(filename);
     Ok(path)
 }
 
-fn dir() -> Result<PathBuf, Error> {
-    let dir = BaseDirs::new()
+/// Ensure that the key-pair storage [dir()] is ready to be used.
+/// Fails with [Error::InsufficientDirPermissions] when said dir
+/// can not be read or can not be created.
+fn dir_ready() -> Result<PathBuf, Error> {
+    let dir = dir();
+    File::open(dir.as_path()).map_err(Error::InsufficientDirPermissions)?;
+    std::fs::create_dir_all(&dir).map_err(WritingError::IO)?;
```

Comment:
Running the following command fails even though  my user has read, write, and execute permissions on `/tmp`.
~~~
XDG_DATA_HOME=/tmp cargo run -p radicle-registry-cli -- key-pair list
~~~

The reason is that `File::open("tmp/radicle-registry-cli")` fails because the directory does not exist yet.

To address this I’d suggest to map the `create_dir_all` error.
```suggestion
    std::fs::create_dir_all(&dir).map_err(|err| Error::CreateDirectoryFailed(err, dir))?;
```

together with
```rust
    #[error("Cannot create directory {1}")]
    CreateDirectoryFailed(#[source] IOError, PathBuf),
```

Reply:
Good catch! We need to check that we can read the directory anyway, so we shouldn't delete the File::open line. If the dir can't be read, neither list neither add will work.

So we could just swap the File::open and the create_dir_all lines and it should fly. I don't feel strong about adding that specific variant, as reporting to the user that the storage directory needs to have read and write access is sufficient.

What do you think?

Reply:
I pushed a fixup adding the new `CannotReadDirectory` and `CannotCreateDirectory` error variants, and swap the lines above to first try to create the dir and then check that it can be read. if it doesn't exist and we don't have permissions to create it, it fails with `CannotCreateDirectory`; if it exists but we can't read it, it fails with `CannotReadDirectory`.